src/routes/assemblies/[id]/members/[groupby]/+page.svelte

```diff
@@ -1,8 +1,243 @@
 <script lang="ts">
+	import Header from '$components/Assemblies/Members/Header.svelte';
+	import Tab from '$components/Assemblies/Members/Tab.svelte';
+	import PoliticianProfile from '$components/PoliticianProfile/PoliticianProfile.svelte';
+	import {
+		Accordion,
+		AccordionItem,
+		Button,
+		Checkbox,
+		FormGroup,
+		Search
+	} from 'carbon-components-svelte';
+	import Minimize from 'carbon-icons-svelte/lib/Minimize.svelte';
+	import SearchLocate from 'carbon-icons-svelte/lib/SearchLocate.svelte';
+	import scrollama from 'scrollama';
+	import { onMount } from 'svelte';
+	import type { PoliticianGroupBy } from './+page.js';
+
 	export let data;
-	$: ({ groups } = data);
+	$: ({ groups, groupByTabs } = data);
+	$: currentPath = groupByTabs.find((e) => e.isActive)?.path ?? '';
+
+	let showFilter = true;
+	let searchQuery = '';
+	let isByDistrict = true;
+	let isByPartylist = true;
+
+	$: formattedSearchQuery = searchQuery.trim();
+	$: filteredGroup =
+		formattedSearchQuery === '' && isByDistrict && isByPartylist
+			? groups
+			: (groups.map((group) => {
+					if ('subgroups' in group) {
+						return {
+							...group,
+							subgroups: group.subgroups.map((subgroup) => {
+								return {
+									...subgroup,
+									members: subgroup.members.filter((member) => {
+										return (
+											(member.firstname + ' ' + member.lastname).includes(formattedSearchQuery) &&
+											((isByDistrict && member.candidateType === 'แบ่งเขต') ||
+												(isByPartylist && member.candidateType === 'บัญชีรายชื่อ'))
+										);
+									})
+								};
+							})
+						};
+					}
+					return group.name.includes(formattedSearchQuery);
+			  }) as PoliticianGroupBy);
+
+	let isMobile = false;
+	let mounted = false;
+	let currentCatg = '';
+	onMount(() => {
+		mounted = true;
+
+		showFilter = window.matchMedia(`(min-width: 672px)`).matches;
+		isMobile = !showFilter;
+		if (showFilter) {
+			currentCatg = (
+				groups[0].name +
+				'-' +
+				('subgroups' in groups[0] ? groups[0].subgroups[0].name : '')
+			).replace(/ /g, '-');
+
+			const scroller = scrollama();
+
+			scroller
+				.setup({
+					step: '.member-subcategory',
```

Comment:
Better avoid global selection. Maybe `bind:this` to the closest parent that contains all the needed elements and query selector inside it.

```ts
step: someParentElement.querySelectorAll('.member-subcategory')
```